content/en/docs/concepts/overview/_index.md

```diff
@@ -120,8 +120,10 @@ Kubernetes provides you with:
   containers onto your nodes to make the best use of your resources.
 * **Self-healing**
   Kubernetes restarts containers that fail, replaces containers, kills containers that don't
-  respond to your user-defined health check, and doesn't advertise them to clients until they
-  are ready to serve.
+  respond to your user-defined health check :- does Kubernetses really act on the container 
+  directly or restart the pod which having that container? kubernetes does not replace the 
```

Comment:
Hi @YashPimple.
I'm not clear on whether the changes improve the existing content.